src/Adapter/Driver/Pgsql/Connection.php

```diff
@@ -124,13 +124,16 @@ public function connect()
         $connection = $this->getConnectionString();
         set_error_handler(function ($number, $string) {
             throw new Exception\RuntimeException(
-                __METHOD__ . ': Unable to connect to database',
+                __CLASS__ . '::connect: Unable to connect to database',
                 null,
                 new Exception\ErrorException($string, $number)
             );
         });
-        $this->resource = pg_connect($connection);
-        restore_error_handler();
+        try {
+            $this->resource = pg_connect($connection);
+        } finally {
+            restore_error_handler();
```

Comment:
I noticed one more bug when I was writing the unit test.

The custom error handler would throw if pg_connect() failed, and restore_error_handler previously **wouldn't get called**.

I'm not sure if connecting to what would normally be an invalid port would work on all platforms (e.g. firewalls), but that's the best idea I had at the time.